### ISSUE TITLE
Cargo does not seem to always propagate `cfg(clippy)` correctly to dependencies

### ISSUE THREAD

**Issue by rmalmain** [rust-lang/rust-clippy#13101]
### Summary

When using path dependencies in `Cargo.toml`, cargo does not seem to propagate the special `cfg(clippy)` to dependencies correctly.
A minimized example [is available here](https://github.com/rmalmain/cargo-bug).

For reference, this issue has been first opened [here](https://github.com/rust-lang/cargo/issues/14251) and then transferred to this repository.

### Reproducer

In the [minimized example repo](https://github.com/rmalmain/cargo-bug), run `cargo clippy --features feature1,feature2`. 

As far as I understand, it should not trigger a compiler error.

### Version

```text
cargo 1.81.0-nightly (154fdac39 2024-07-07)
release: 1.81.0-nightly
commit-hash: 154fdac39ae9629954e19e9986fd2cf2cdd8d964
commit-date: 2024-07-07
host: x86_64-unknown-linux-gnu
libgit2: 1.8.1 (sys:0.19.0 vendored)
libcurl: 8.8.0-DEV (sys:0.4.73+curl-8.8.0 vendored ssl:OpenSSL/1.1.1w)
ssl: OpenSSL 1.1.1w  11 Sep 2023
os: Arch Linux Rolling Release [64-bit]
```

**Comment by Jarcho** [comment]
`cfg(clippy)` is only set for the crates clippy is run on, not dependencies. This is the same for the deprecated `cfg(feature = "cargo-clippy")`.

**Comment by rmalmain** [comment]
Thank you for the clarification!

If this is the intended behavior, is there a way to enable `cfg(clippy)` for dependencies as well?
For now, we solved this by creating a dummy feature that we propagate during clippy runs, but I believe this is definitely not a good approach.

**Comment by Alexendoo** [comment]
Clippy gets run for dependencies that are in the workspace, so adding `[workspace]` to the root package would be enough since it automatically adds path deps to the workspace. I don't think we have this well documented

**Comment by rmalmain** [comment]
I can confirm it indeed solved the issue, thank you!
I have not been able to see this information in the documentation so far.

It would be very helpful to add a few words about this [around here](https://doc.rust-lang.org/clippy/configuration.html#disabling-evaluation-of-certain-code) IMHO.
